analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/pom/DataExtractor.java

```diff
@@ -115,6 +115,45 @@ public String extractPackagingType(String groupId, String artifactId, String ver
         return packaging;
     }
 
+    /**
+     * Extracts project name from POM of certain Maven coordinate.
+     *
+     * @param groupId    groupId of the coordinate
+     * @param artifactId artifactId of the coordinate
+     * @param version    version of the coordinate
+     * @return Extracted project name as String
+     */
+    public String extractProjectName(String groupId, String artifactId, String version) {
+        String name = null;
+        try {
+            ByteArrayInputStream pomByteStream;
+            if ((groupId + ":" + artifactId + ":" + version).equals(this.mavenCoordinate)) {
+                pomByteStream = new ByteArrayInputStream(this.pomContents.getBytes());
+            } else {
+                pomByteStream = new ByteArrayInputStream(
+                        this.downloadPom(artifactId, groupId, version)
+                                .orElseThrow(FileNotFoundException::new).getBytes());
+            }
```

Comment:
I would have replaced this part with a ternary operator. It looks cleaner and more readable. Same goes for the similar part in other `extract[...]`methods.

Reply:
Fixed that